provider/runbook_resource_test.go

```diff
@@ -286,19 +325,38 @@ func testAccCheckRunbookResourceExistsWithAttributes_update(resourceName string)
 			return fmt.Errorf("Unexpected owner ID. Expected:%s, got: %s", expected, got)
 		}
 
-		if len(runbookResponse.Steps) != 1 {
-			return fmt.Errorf("Unexpected number of steps. Expected 1 step, got: %v", len(runbookResponse.Steps))
+		if len(runbookResponse.Steps) != 2 {
+			return fmt.Errorf("Unexpected number of steps. Expected 2 steps, got: %v", len(runbookResponse.Steps))
 		}
 
 		for index, step := range runbookResponse.Steps {
 			key := fmt.Sprintf("steps.%d", index)
 			if runbookResource.Primary.Attributes[key+".name"] != step.Name {
 				return fmt.Errorf("Unexpected runbook step name. Expected %s, got: %s", step.Name, runbookResource.Primary.Attributes[key+".name"])
 			}
+
 			if runbookResource.Primary.Attributes[key+".action_id"] != step.ActionID {
 				return fmt.Errorf("Unexpected runbook step action_id. Expected %s, got: %s", step.Name, runbookResource.Primary.Attributes[key+".action_id"])
 			}
-			// TODO: Test that config matches
+
+			var config []byte
+			if len(step.Config) > 0 {
+				config, err = json.Marshal(step.Config)
+				if err != nil {
+					return fmt.Errorf("Unexpected error converting runbook step config to JSON: %v", err)
+				}
+			}
+			if runbookResource.Primary.Attributes[key+".config"] != string(config) {
+				return fmt.Errorf("Unexpected runbook step config. Expected %s, got: %s", step.Config, runbookResource.Primary.Attributes[key+".config"])
+			}
+
+			if runbookResource.Primary.Attributes[key+".repeats"] != fmt.Sprintf("%t", step.Repeats) {
```

Comment:
Oh nice assuming this will give a more detailed error around which step key actually failed.

Reply:
that's what i'm hoping 🤞🏼 i tried to get it to fail just so i could confirm that it really does what i think it should but i couldn't get it to work.